pumpkin-protocol/src/client/play/c_entity_metadata.rs

```diff
@@ -5,13 +5,13 @@ use crate::VarInt;
 
 #[derive(Serialize)]
 #[packet(0x58)]
-pub struct CSetEntityMetadata<T> {
-    entity_id: VarInt,
-    metadata: Metadata<T>,
-    end: u8,
+pub struct CSetEntityMetadata<T: Serialize> {
+    pub entity_id: VarInt,
+    pub metadata: Metadata<T>,
+    pub end: u8,
 }
```

Comment:
I see that you like to make the fields public, But its more of an "interal convention" in pumpkin to use the new function